### PR TITLE
fix(apply_filters): distinct between has_many and has_one

### PR DESCRIPTION
**Pull request by Ashiroes**

### CODE REVIEW

**Comment by pauldub** [comment]
@Ashiroes could you explain the changes a little better?

It may be better to introduce a new parameter allowing one to specify which association is to be queried. 

This would be useful when a model has multiple associations to another one.

File: lib/fetcheable_on_api/filterable.rb [hunk 1]
```diff
@@ -113,7 +113,15 @@ def apply_filters(collection)
         predicate = filters_configuration[column.to_sym].fetch(:with, :ilike)
 
         if collection_klass != klass
-          collection = collection.joins(klass.table_name.to_sym)
+          if collection_klass.reflect_on_association(
+            klass.name.pluralize.downcase.to_sym
+          )
+            collection = collection.joins(klass.name.pluralize.downcase.to_sym)
```
Comment (fabienpiette) [inline lib/fetcheable_on_api/filterable.rb]: In multi-line you can use this syntax:
```ruby
if foo(
  bar
)
then 
  ...
elsif bar(
  foo
)
then
 ...
end
```

File: lib/fetcheable_on_api/filterable.rb [hunk 2]
```diff
@@ -113,7 +113,15 @@ def apply_filters(collection)
         predicate = filters_configuration[column.to_sym].fetch(:with, :ilike)
 
         if collection_klass != klass
-          collection = collection.joins(klass.table_name.to_sym)
+          if collection_klass.reflect_on_association(
+            klass.name.pluralize.downcase.to_sym
+          )
```
Comment (fabienpiette) [inline lib/fetcheable_on_api/filterable.rb]: You can use `.present?` it's easier to read

File: lib/fetcheable_on_api/filterable.rb [hunk 3]
```diff
@@ -113,7 +113,15 @@ def apply_filters(collection)
...
+            collection = collection.joins(klass.name.pluralize.downcase.to_sym)
+          elsif collection_klass.reflect_on_association(
+            klass.name.downcase.to_sym
+          )
+            collection = collection.joins(klass.name.downcase.to_sym)
+          end
```
Comment (fabienpiette) [inline lib/fetcheable_on_api/filterable.rb]: for clarity you can move this code into a method

**Comment by pauldub** [comment]
Closed in favor of #22

**Pull request closed without merging**